lxd/api.go

```diff
@@ -71,7 +72,24 @@ func restServer(d *Daemon) *http.Server {
 	uiEnabled := uiPath != "" && shared.PathExists(uiPath)
 	if uiEnabled {
 		uiHTTPDir := uiHTTPDir{http.Dir(uiPath)}
-		mux.PathPrefix("/ui/").Handler(http.StripPrefix("/ui/", http.FileServer(uiHTTPDir)))
+
+		// Serve the LXD user interface.
+		uiHandler := http.StripPrefix("/ui/", http.FileServer(uiHTTPDir))
+
+		// Set security headers
+		uiHandlerWithSecurity := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			w.Header().Set("Permissions-Policy", "interest-cohort=()")
+			w.Header().Set("X-Content-Type-Options", "nosniff")
+			w.Header().Set("X-Frame-Options", "SAMEORIGIN")
+			w.Header().Set("Content-Security-Policy", "default-src 'self'; script-src 'self'; object-src 'none'")
```

Comment:
I don't have a hard requirement for the CSP.

Did some more testing, this would be needed, though OIDC is still not working for me with it. That needs some more adjustments:

```
w.Header().Set("Content-Security-Policy", "default-src 'self'; script-src 'self' 'unsafe-inline'; object-src 'self' https://cloud-images.ubuntu.com; font-src 'self' https://assets.ubuntu.com; img-src 'self' data: https://assets.ubuntu.com; style-src 'self' 'unsafe-inline'")
```

Reply:
Lets remove CSP for now so we can get some of this into the LTS.

Reply:
Marking this as fixed